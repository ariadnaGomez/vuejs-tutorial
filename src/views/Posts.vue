<template>
  <div class="posts">
    <pre>{{ itemsArray }}</pre>
    <el-button @click="addCheckbox">AÃ±adir checkbox</el-button>
    <checkbox-item
      v-for="item in itemsArray"
      :key="item.id"
      :id="item.id"
      :title="item.title"
      :is-active="item.isActive"
      class="test"
      @checkbox-clicked="saveItem"
    />
    <br>
    <br>
    <el-button @click="goCheckboxList">Ver listado de checkbox</el-button>
  </div>
</template>

<script>
import CheckboxItem from '../components/checkbox-item.vue'
import {mapGetters, mapMutations} from 'vuex'

export default {
  name: 'Posts',
  components: {CheckboxItem},
  data () {
    return {
      itemsArray: [
        {
          id: 0,
          isActive: false,
          title: 'titulo0'
        },
        {
          id: 1,
          isActive: false,
          title: 'titulo1'
        },
        {
          id: 2,
          isActive: false,
          title: 'titulo2'
        },
        {
          id: 3,
          isActive: false,
          title: 'titulo3'
        },
        {
          id: 4,
          isActive: false,
          title: 'titulo4'
        }],
      index: 5
    }
  },
  computed: {
    ...mapGetters(['getIsActive', 'getCheckboxListAux'])
  },
  created: function () {
    if (this.getCheckboxListAux.length > 0) {
      this.itemsArray = this.getCheckboxListAux
    }
    for (var i = 0; i < this.itemsArray.length; i += 1) {
      this.itemsArray[i].isActive = this.getIsActive(i)
    }
  },
  destroyed: function () {
    this.setCheckboxAux(this.itemsArray)
  },
  methods: {
    ...mapMutations(['addNewCheckbox', 'setCheckboxAux']),
    saveItem (checkboxItem) {
      this.addNewCheckbox(checkboxItem)
    },
    goCheckboxList () {
      this.$router.push('CheckboxList')
    },
    addCheckbox () {
      const newItem = {
        id: this.index,
        isActive: false,
        title: 'titulo' + this.index
      }
      this.itemsArray.push(newItem)
      this.index++
    }
  }
}

</script>
<style></style>
