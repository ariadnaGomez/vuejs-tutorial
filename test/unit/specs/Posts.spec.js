import { shallow, mount } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'
import Posts from '@/views/Posts.vue'
import CheckboxItem from '@/components/checkbox-item.vue'

describe('Posts', () => {
  test('should render a checkbox item for each item', () => {
    const wrapper = shallow(Posts)
    const lengthArrayItems = wrapper.vm.itemsArray.length
    expect(wrapper.findAll(CheckboxItem).length).toBe(lengthArrayItems)
  })

  test('should save each checkbox in the store', () => {
    const wrapper = shallow(Posts)
    const itemCheckbox = wrapper.find(CheckboxItem)
    const contentItemArray = { isActive: true, id: 1 }
    wrapper.setData({ itemsArray:
      [{ isActive: false, id: 1 }, {isActive: false, id: 2}] })
    itemCheckbox.vm.$emit('checkbox-clicked', contentItemArray)
    wrapper.update()
    expect(wrapper.vm.itemsArray[0].isActive).toBe(contentItemArray.isActive)
  })

  it('has same HTML structure', () => {
    const renderer = createRenderer()
    const wrapper = mount(Posts)
    wrapper.setData({ itemsArray:
      [{ isActive: false, id: 1 }, {isActive: false, id: 2}] })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
