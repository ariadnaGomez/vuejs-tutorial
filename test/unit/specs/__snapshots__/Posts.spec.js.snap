// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts has same HTML structure 1`] = `
"<div data-server-rendered=\\"true\\" class=\\"posts\\"><pre>[
  {
    &quot;isActive&quot;: false,
    &quot;id&quot;: 1
  },
  {
    &quot;isActive&quot;: false,
    &quot;id&quot;: 2
  }
]</pre> <div class=\\"container test\\"><p></p> <input type=\\"checkbox\\">
  No esta activo
</div><div class=\\"container test\\"><p></p> <input type=\\"checkbox\\">
  No esta activo
</div></div>"
`;
